docs/modules/ROOT/pages/api/erc20.adoc

```diff
@@ -0,0 +1,372 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+:eip20: https://eips.ethereum.org/EIPS/eip-20[EIP-20]
+:erc20-guide: xref:erc20.adoc[ERC20 guide]
+
+
+= ERC20
+
+Reference of interfaces and utilities related to ERC20 contracts.
+
+TIP: For an overview of ERC20, read our {erc20-guide}.
+
+== Core
+
+[.contract]
+[[IERC20]]
+=== `++IERC20++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/token/erc20/interface.cairo#L6-L19[{github-icon},role=heading-link]
+
+[.hljs-theme-light.nopadding]
+```javascript
+use openzeppelin::token::erc20::interface::IERC20;
+```
+
+Interface of the IERC20 standard as defined in {eip20}.
+
+[.contract-index]
+.Functions
+--
+* xref:#IERC20-total_supply[`++total_supply()++`]
+* xref:#IERC20-balance_of[`++balance_of()++`]
+* xref:#IERC20-allowance[`++allowance(owner: ContractAddress, spender: ContractAddress)++`]
+* xref:#IERC20-transfer[`++transfer(recipient: ContractAddress, amount: u256)++`]
+* xref:#IERC20-transfer_from[`++transfer_from(sender: ContractAddress, recipient: ContractAddress, amount: u256)++`]
+* xref:#IERC20-approve[`++approve(spender: ContractAddress, amount: u256)++`]
```

Comment:
i thought we were skipping argument types in here ([like in Solidity](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20)). although i wouldn't skip return values

Reply:
for details I agree, but to me return value is as important as the arguments

Reply:
Sounds good. No strong opinions on whether we should include the return values. I just followed the [Solidity docs](https://docs.openzeppelin.com/contracts/4.x/api/token/erc20#IERC20) approach. Happy to change it if we really think it's necessary

Reply:
let's not block this PR and discuss it in #726 instead

Reply:
> for details I agree, but to me return value is as important as the arguments

Whoops, I missed this in my response^

> let's not block this PR and discuss it in https://github.com/OpenZeppelin/cairo-contracts/issues/726 instead

Sounds good